src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -26,6 +31,188 @@
 from WMCore.DAOFactory import DAOFactory
 
 
+def findFiles(path, pat):
+    """
+    Find files within given path and matching given pattern.
+    :param path: starting directory path
+    :param pat: match pattern, e.g. pileup name
+    :return: file name of sandbox tarball
+    """
+    files = []
+    for idir, _, _ in os.walk(path):
+        files.extend(glob.glob(os.path.join(idir, pat)))
+    return files
+
+
+def tarMode(tfile, op):
+    """
+    Extract proper mode of operation for given tar file. For instance,
+    if op='r' and tfile name is file.tar.gz we should get 'r:gz',
+    while if tfile name is file.tar.bz2 we should get 'r':bz2', while
+    if tfile name is file.tar we should get 'r', etc.
+    :param op: mode of operation, e.g. 'r', or 'w'
+    :param tfile: sandbox tar file
+    :return: mode of operation
+    """
+    ext = tfile.split(".")[-1]
+    if ext == 'tar':
+        return op
+    mode = op + ':' + ext
+    return mode
+
+
+def findJsonSandboxFiles(tfile):
+    """
+    Find location of sandbox JSON files
+    :param tfile: sandbox tar file
+    :return: list of file names
+    """
+    files = []
+    with tarfile.open(tfile, tarMode(tfile, 'r'), encoding='utf-8') as tar:
+        for tarInfo in tar.getmembers():
+            if tarInfo.name.endswith("pileupconf.json"):
+                files.append(tarInfo.name)
+    return files
+
+
+def extractPileupconf(tfile, fname):
+    """
+    Extract content of given file name from sandbox tar file
+    :param tfile: sandbox tar file
+    :param fnamne: name of the file to extract
+    :return: content of file
+    """
+    with tarfile.open(tfile, tarMode(tfile, 'r'), encoding='utf-8') as tar:
```

Comment:
Wouldn't it be more clear if we move the `tarMode(tfile, 'r')` to the line above and get it assigned to a variable, then use the variable in this tarfile.open()? Really a minor comment, but I feel like it would be nice to explicitly separate those 2 actions.

Reply:
done